jinahub/rankers/CatboostRanker/catboost_ranker.py

```diff
@@ -0,0 +1,167 @@
+        return np.array(feature_vectors), label_vector, group_ids
+
+    def _extract_weights(self, docs: DocumentArray):
+        """Extract weights data from docs, weights are needed if the query_docs has different
+        importance. Once given, we'll get values stored in the Document.tags based on the field name.
+        """
+        weight_vector = []
+        for doc in docs:
+            for _ in doc.matches:
+                # weight need to have the same size of matches, while get from doc.
+                weight_vector.append(doc.tags.get(self.weight))
+        return weight_vector
+
+    def build_catboost_pool(self, docs: DocumentArray):
+        """Create the catboost :class:`Pool` for training.
+
+        :param docs: A :class:`DocumentArray` to build the pool.
+        :return: A :class:`Pool` object of catboost.
+        """
+        data, label, group_id = self._extract_features(docs)
+        if self.weight:
+            return Pool(
+                data=data,
+                label=label,
+                group_id=group_id,
+                weight=self._extract_weights(docs),
+            )
+        else:
+            return Pool(data=data, label=label, group_id=group_id)
+
+    @requests(on='/train')
+    def train(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """This endpoint fits a catboost ranker based on docs.
+
+        :param docs: Documents to fit the model, each docs should have fields corresponded to
+          the :attr:`query_features` and has matches with corresponded :attr:`match_features` to
+          extract feature from. Also, a label field should be stored in each match document.
+        :param parameters: Parameters to set, we'll get :attr:`catboost_parameters` from the parameters field.
+        """
+        catboost_parameters = parameters.get(
+            'catboost_parameters', self.catboost_parameters
+        )
+        train_pool = self.build_catboost_pool(docs)
+        self.model = CatBoostRanker(**catboost_parameters)
+        self.model.fit(train_pool)
+
+    @requests(on='/search')
+    def rank(self, docs: DocumentArray, parameters: Dict = {}, **kwargs):
+        """This endpoint ranks a catboost ranker based on docs.
+
+        :param docs: Documents to fit the model, each docs should have fields corresponded to
+          the :attr:`query_features` and has matches with corresponded :attr:`match_features` to
+          extract feature from. An additional field will be attached to the matched documents as was defined
+          as :attr:`label`.
+        :param parameters: Parameters to set, we'll get :attr:`catboost_parameters` from the parameters field.
+        """
+        feature_vecs, _, _ = self._extract_features(docs)
+        if not self.model.is_fitted():
+            raise ValueError(
+                'You need to train your model before make prediction, call `train` endpoint first.'
```

Comment:
```suggestion
                'You need to train your model before making predictions, call the `train` endpoint first.'
```

Reply:
This is not true, U do not need to call `train` endpoint first. U need to provide a fitted model, even trained offline or calling training endpoint